File: backend/app/modules/subscriptions/permissions.py
from app.modules.identity.enums import PermissionKey

SUBSCRIPTION_PERMISSIONS = {
  PermissionKey.SUBSCRIPTION_READ:
    "View organization subscription and usage information.",

  PermissionKey.SUBSCRIPTION_MANAGE:
    "Manage organization subscription.",

  PermissionKey.SUBSCRIPTION_BILLING_MANAGE:
    "Manage organization billing information and subscription changes.",
}

File: frontend/src/modules/subscriptions/permissions.ts
export const SUBSCRIPTION_PERMISSIONS = {
  SUBSCRIPTION_READ: "subscription:read",
  SUBSCRIPTION_MANAGE: "subscription:manage",
  SUBSCRIPTION_BILLING_MANAGE: "subscription:billing_manage",
} as const;

export function hasSubscriptionPermission(
  permissions: string[],
  permission: string,
): boolean {
  return permissions.includes(permission);
}